mm2src/coins/lightning.rs

```diff
@@ -1323,11 +1342,94 @@ impl MmCoin for LightningCoin {
 
     fn mature_confirmations(&self) -> Option<u32> { None }
 
-    // Todo: This uses default data for now for the sake of swap P.O.C., this should be implemented probably when implementing order matching if it's needed
-    fn coin_protocol_info(&self) -> Vec<u8> { Vec::new() }
+    // Channels for users/non-routing nodes should be private, so routing hints are sent as part of the protocol info
+    // alongside the receiver lightning node address/pubkey.
+    // Note: This is required only for the side that's getting paid in lightning.
+    // Todo: should take in consideration JIT routing and using LSPs in next PRs
+    fn coin_protocol_info(&self, amount_to_receive: Option<MmNumber>) -> Vec<u8> {
+        let amt_msat = match amount_to_receive.map(|a| sat_from_big_decimal(&a.into(), self.decimals())) {
+            Some(Ok(amt)) => amt,
+            Some(Err(e)) => {
+                error!("{}", e);
+                return Vec::new();
+            },
```

Comment:
If the amount can't be converted to `u64` it will be catched in previous balance checks anyways. Returning an error from this function for this conversion error will be overdoing this.